core/tests/unit/kmnkbd/test_actions_normalize.cpp

```diff
@@ -329,5 +592,25 @@ int main(int argc, char *argv []) {
   arg_path = argv[arg_color ? 2 : 1];
 #endif
 
-  run_tests();
+  run_actions_normalize_tests();
+  run_actions_update_app_context_nfu_tests();
+}
+
+
+void compare_context(km_core_context *app_context, const km_core_cp* expected_final_app_context) {
+  // Compare context items -- to ensure no markers have leaked into app context
+  km_core_context_item *actual_final_app_context_items = nullptr, *expected_final_app_context_items = nullptr;
+  try_status(km_core_context_get(app_context, &actual_final_app_context_items));
+  try_status(context_items_from_utf16(expected_final_app_context, &expected_final_app_context_items));
```

Comment:
This will strip the markers when `expected_final_app_context_items`. Which when I think about it is what you want as you want to make sure the app_context does not have any markers.